File: core/src/com/mygdx/game/Jetrix.java
package com.mygdx.game;

import com.badlogic.gdx.ApplicationAdapter;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.GL20;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.mygdx.game.states.GameStateManager;
import com.mygdx.game.states.PlayState;

public class Jetrix extends ApplicationAdapter {

	public static final int WIDTH = 480;
	public static final int HEIGHT = 800;

	public static final String TITLE = "Jetrix";

	private SpriteBatch batch;
	private GameStateManager gsm;
	
	@Override
	public void create () {
		batch = new SpriteBatch();
		gsm = new GameStateManager();
		gsm.push(new PlayState());
	}

	@Override
	public void render () {
		gsm.update(Gdx.graphics.getDeltaTime());
		Gdx.gl.glClearColor(1, 0, 0, 1);
		Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
		gsm.render(batch);
	}
	
	@Override
	public void dispose () {
		super.dispose();
		batch.dispose();
	}
}

File: core/src/com/mygdx/game/Jetrix.kt
package com.mygdx.game

import com.badlogic.ashley.core.Engine
import com.badlogic.gdx.Game
import com.badlogic.gdx.graphics.OrthographicCamera
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.utils.Logger
import com.mygdx.game.screens.PlayScreen

class Jetrix : Game() {

    lateinit var spriteBatch: SpriteBatch
    lateinit var camera: OrthographicCamera

    lateinit var engine: Engine

    companion object {
        val logger = Logger("Jetrix")
    }

    override fun create() {
        spriteBatch = SpriteBatch()
        camera = OrthographicCamera(Config.width.toFloat(), Config.height.toFloat())

        engine = Engine()

        setScreen(PlayScreen(this))
    }

}